pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotInstanceAssignmentRestletResource.java

```diff
@@ -191,6 +190,22 @@ public Map<InstancePartitionsType, InstancePartitions> assignInstances(
     return instancePartitionsMap;
   }
 
+  /**
+   * Assign instances given the type of instancePartitions.
+   * @param instancePartitionsMap the empty map to be filled.
+   * @param tableConfig table config
+   * @param instanceConfigs list of instance configs
+   * @param instancePartitionsType type of instancePartitions
+   */
+  private void assignInstancesForInstancePartitionsType(
+      Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap, TableConfig tableConfig,
+      List<InstanceConfig> instanceConfigs, InstancePartitionsType instancePartitionsType) {
+    InstancePartitions existingInstancePartitions = InstancePartitionsUtils
+        .fetchOrComputeInstancePartitions(_resourceManager.getHelixZkManager(), tableConfig, instancePartitionsType);
```

Comment:
We should not compute the instance partitions if it does not exist. The `assignInstances()` should be able to handle the case of not having an existing instance partitions

Reply:
Updated.